sw/device/silicon_creator/manuf/skus/earlgrey_a0/sival_bringup/ft_personalize.c

```diff
@@ -40,40 +41,64 @@
-              "struct should match the value of `kCdi1MaxCertSizeBytes`.");
-static_assert(kTpmEkMaxTbsSizeBytes == 844,
-              "The `tpm_ek_tbs_certificate` buffer size in the `manuf_certs_t`"
-              "struct should match the value of `kTpmEkMaxTbsSizeBytes`.");
-static_assert(kTpmCekMaxTbsSizeBytes == 456,
-              "The `tpm_cek_tbs_certificate` buffer size in the `manuf_certs_t`"
-              "struct should match the value of `kTpmCekMaxTbsSizeBytes`.");
-static_assert(kTpmCikMaxTbsSizeBytes == 456,
-              "The `tpm_cik_tbs_certificate` buffer size in the `manuf_certs_t`"
-              "struct should match the value of `kTpmCikMaxTbsSizeBytes`.");
-static_assert(kUdsMaxCertSizeBytes == 818,
-              "The `uds_tbs_certificate` buffer size in the `manuf_certs_t` "
-              "struct should match the value of `kUdsMaxTbsSizeBytes`.");
-static_assert(kTpmEkMaxCertSizeBytes == 935,
-              "The `tpm_ek_certificate` buffer size in the `manuf_certs_t` "
-              "struct should match the value of `kTpmEkMaxCertSizeBytes`.");
-static_assert(kTpmCekMaxCertSizeBytes == 547,
-              "The `tpm_cek_certificate` buffer size in the `manuf_certs_t` "
-              "struct should match the value of `kTpmCekMaxCertSizeBytes`.");
-static_assert(kTpmCikMaxCertSizeBytes == 547,
-              "The `tpm_cik_certificate` buffer size in the `manuf_certs_t` "
-              "struct should match the value of `kTpmCikMaxCertSizeBytes`.");
+enum {
+  /**
+   * Alignment mask for aligning on 32-bit words.
+   */
+  k32BitWordAlignMask = 0x3u,
+
+  /**
+   * TBS certificate buffer sizes.
+   */
+  kUdsTbsCertBufferSizeInBytes =
+      (kUdsMaxTbsSizeBytes + k32BitWordAlignMask) & ~k32BitWordAlignMask,
+  kTpmEkTbsCertBufferSizeInBytes =
+      (kTpmEkMaxTbsSizeBytes + k32BitWordAlignMask) & ~k32BitWordAlignMask,
+  kTpmCekTbsCertBufferSizeInBytes =
+      (kTpmCekMaxTbsSizeBytes + k32BitWordAlignMask) & ~k32BitWordAlignMask,
+  kTpmCikTbsCertBufferSizeInBytes =
+      (kTpmCikMaxTbsSizeBytes + k32BitWordAlignMask) & ~k32BitWordAlignMask,
+
+  /**
+   * Endorsed certificate buffer sizes.
+   */
+  kUdsCertBufferSizeInBytes =
+      (kUdsMaxCertSizeBytes + k32BitWordAlignMask) & ~k32BitWordAlignMask,
+  kCdi0CertBufferSizeInBytes =
+      (kCdi0MaxCertSizeBytes + k32BitWordAlignMask) & ~k32BitWordAlignMask,
+  kCdi1CertBufferSizeInBytes =
+      (kCdi1MaxCertSizeBytes + k32BitWordAlignMask) & ~k32BitWordAlignMask,
+  kTpmEkCertBufferSizeInBytes =
+      (kTpmEkMaxCertSizeBytes + k32BitWordAlignMask) & ~k32BitWordAlignMask,
+  kTpmCekCertBufferSizeInBytes =
+      (kTpmCekMaxCertSizeBytes + k32BitWordAlignMask) & ~k32BitWordAlignMask,
+  kTpmCikCertBufferSizeInBytes =
+      (kTpmCikMaxCertSizeBytes + k32BitWordAlignMask) & ~k32BitWordAlignMask,
+};
+
+// Check TBS certificate buffer sizes.
+OT_ASSERT_MEMBER_SIZE_AS_ENUM(manuf_certs_t, uds_tbs_certificate,
```

Comment:
I actually meant to ask earlier: why are these asserts needed in the first place? Shouldnt there just be a check when certs are stored that they fit, and another check confirming their validity?

The size of the signed cert is not guaranteed - signature R and S components are expected to be 32 bytes, but could be up to 33 in case there is a need to make sure the value is positive, or shorter than 32 if there are leading zeros...

Reply:
The static asserts are needed because we need to hard code the sizes of the struct fields in the UJSON certificate struct definitions in `provisioning_data.h` (due to limitations with the UJSON framework), and we want to ensure these struct definitions stay in sync with the autogenerated enum constants that get generated from the HJSON by the ot_certs framework.